src/Components/EnvironmentPicker/EnvironmentPicker.tsx

```diff
@@ -122,454 +115,462 @@ const EnvironmentPicker = (props: EnvironmentPickerProps) => {
-                return urlObj.hostname.split('.')[0] + urlObj.pathname; // i.e. AzureStorageAccountName/ContainerName
-            } else {
-                return t('environmentPicker.noContainer');
-            }
+    const displayTextForContainer = useCallback(
+        (container: string | IAzureResource) => {
+            const displayText = EnvironmentPickerManager.displayTextForContainer(
+                container,
+                storageAccountItems.selectedStorageAccount
+            );
+            return displayText || t('environmentPicker.noContainer');
         },
-        [t]
+        [t, storageAccountItems.selectedStorageAccount]
     );
 
+    const handleOnEnvironmentResourceChange = (
+        resource: IAzureResource | string,
+        resources: Array<IAzureResource | string>
+    ) => {
+        setEnvironmentItems({
+            ...environmentItems,
+            environmentToEdit: resource,
+            environments: resources
+        });
+    };
+
+    const handleOnStorageAccountResourceChange = (
+        resource: IAzureResource | string,
+        resources: Array<IAzureResource | string>
+    ) => {
+        setStorageAccountItems({
+            ...storageAccountItems,
+            storageAccountToEdit: resource,
+            storageAccounts: resources
+        });
+
+        // when changing the storage account, update the containers from default mappings until containers are fetched
+        setContainerItems({
+            ...containerItems,
+            containers:
+                defaultStorageAccountToContainersMappingRef.current?.find(
+                    (mapping) =>
+                        EnvironmentPickerManager.isResourceUrlsEqual(
+                            mapping.storageAccountUrl,
+                            EnvironmentPickerManager.getResourceUrl(
+                                resource,
+                                AzureResourceTypes.StorageAccount
+                            )
+                        )
+                )?.containerNames || [],
+            containerToEdit: null
+        });
+        hasFetchedResources.current.storageBlobContainers = false; // reset this flag as we click on different storage account since it fetches containers again with different storage account id
+    };
+
+    const handleOnStorageContainerResourceChange = (
+        resource: IAzureResource | string,
+        resources: Array<IAzureResource | string>
+    ) => {
+        setContainerItems({
+            ...containerItems,
+            containerToEdit: resource,
+            containers: resources
+        });
+
+        // update mappings in case change is based on addition or removal of a resource
+        const defaultStorageAccountMapping = defaultStorageAccountToContainersMappingRef.current?.find(
+            (mapping) =>
+                EnvironmentPickerManager.isResourceUrlsEqual(
+                    EnvironmentPickerManager.getResourceUrl(
```

Comment:
since EnvironmentPickerManager is an internal manager class I didn't want to export the methods in there to surface (it is not like Utils class where other components or other src files also need those methods). I can export those though. What do you think @msnyder-msft?

Reply:
I mean I don't think anyone is going to import a function from a utility class of a specific component so I'm not sure there's really a risk there. Are you just trying to hide it from intellisense?

Reply:
yep it was my concern, but revised it as you suggested